server/events/project_command_context_builder.go

```diff
@@ -257,7 +263,8 @@ func newProjectCommandContext(ctx *CommandContext,
 		Workspace:                 projCfg.Workspace,
 		PolicySets:                policySets,
 		Tags:                      projCfg.Tags,
-		PullReqStatus:              pullStatus,
+		PullReqStatus:             pullStatus,
+		JobID:                     jobID,
```

Comment:
Should we always create a `jobID`? You should just generate it in this function, even if we don't stream anything for a specific project context, it should still have job id.

Reply:
So you mean instead of allowing the calling function to control what the job ID is, just pass the `JobIDGenerator` to this function  `newProjectCommandContext()` and call `Generate()` when assigning jobID in the models.ProjectCommandContext?

Reply:
I don't see why we need to even bother with passing it in as an argument and not just directly call it in here? I think `JobIDGenerator` interface is no longer needed, it is only used in here anyway. So we can remove the whole dependency chain.

Reply:
Yup, I've removed the dependency chain. However, one drawback of generating it in the method is we can't mock the value of the job ID in the tests. So, before comparing the values, we have to set it to an arbitrary value https://github.com/lyft/atlantis/blob/388d98f330ad952acf03177e586bd7b242541fae/server/events/project_command_builder_internal_test.go#L663-L665
However, I also think it's worth removing the unnecessary dependency chain for a trivial operation :)